server/src/com/thoughtworks/go/config/GoConfigDao.java

```diff
@@ -624,7 +624,42 @@ public CruiseConfig update(CruiseConfig cruiseConfig) throws Exception {
         }
     }
 
+    public static class UpdateEnvironmentCommand implements UpdateConfigCommand {
+
+        private final String environmentToUpdate;
+        private final EnvironmentConfig newEnvironmentConfig;
+
+        public UpdateEnvironmentCommand(String environmentToUpdate, EnvironmentConfig newEnvironmentConfig) {
```

Comment:
We can modify the name of the existing environment,
So if the name has modified, the old name will be used to get the existing copy of the environment and in this case `environmentToUpdate != newEnvironmentConfig.name`